owsi-core/owsi-core-components/owsi-core-component-jpa-more/src/main/java/fr/openwide/core/jpa/more/business/history/service/HistoryValueServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package fr.openwide.core.jpa.more.business.history.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import fr.openwide.core.commons.util.rendering.IRenderer;
+import fr.openwide.core.jpa.business.generic.model.GenericEntity;
+import fr.openwide.core.jpa.business.generic.model.GenericEntityReference;
+import fr.openwide.core.jpa.business.generic.service.IEntityService;
+import fr.openwide.core.jpa.more.business.history.model.embeddable.HistoryEntityReference;
+import fr.openwide.core.jpa.more.business.history.model.embeddable.HistoryValue;
+import fr.openwide.core.jpa.more.rendering.service.IRendererService;
+import fr.openwide.core.jpa.util.HibernateUtils;
+import fr.openwide.core.spring.property.SpringPropertyIds;
+import fr.openwide.core.spring.property.service.IPropertyService;
+
+@Service
+public class HistoryValueServiceImpl implements IHistoryValueService {
```

Comment:
This, and in particular the `serialize` method, should be override-able in projects.